features/unattached_commands.feature

```diff
@@ -0,0 +1,63 @@
+Feature: Command behaviour when unattached
```

Comment:
Let me know if I am getting hung up on something, but the filename caused me to get stuck on how we grow these test cases.

So the cases we have will be the set of (platforms * releases) right? should our features files be broken out by release as well? (e.g. unattached_trusty.feature)?

Trying to think through the test cases we would eventually get:
unattached_{trusty, xenial, bionic}
attach_{trusty, xenial, bionic}
status_{trusty, xenial, bionic}
esm_infra_{trusty, xenial, bionic}
livepatch_{trusty, xenial, bionic}
...

And inside each put the platform specific code?

Reply:
This may be another case (like VM vs container) where we don't want to capture the dimensions within the test code, but in the way we run the tests. So, similar to my suggestion at the bottom of https://github.com/CanonicalLtd/ubuntu-advantage-client/pull/767#issuecomment-542834304, we could have:

* "Given an Ubuntu system" (run on all Ubuntu releases), and
* "Given a trusty system" (run only on trusty, skip on all others).

So I think we would then have `platforms × releases` common Given definitions (with each set also including an "all of them" option). So we would have:

* Given an Ubuntu system
* Given an Ubuntu lxd container
* Given an Ubuntu VM
* Given a trusty system
* Given a trusty lxd container
* Given a trusty VM
* ... these last three repeated for each Ubuntu release (but implemented with pattern matching for the steps, so we won't have to implement this N*M different times) ...

And for tests which we want to run on a subset of releases, we could do something like (completely untested!):

```gherkin
Scenario Outline: example scenario
  Given a <release_name> system
  When I check the release version
  Then I see that it's <version>

  Examples: Ubuntu Releases
    | release_name | version |
    | xenial       | 16.04   |
    | bionic       | 18.04   |
```

to minimise repetition.

Reply:
@powersj I don't think there's anything concrete to do until we're introducing our first new {platform, release}, does the above make enough sense to you for us to proceed?

Reply:
agreed